source-academy/frontend
# Added explicit control as a source 4 variant

### Description
This PR adds Source 4 Explicit Control Evaluator as a variant. 
It requires [source-academy/js-slang/pull/1361](https://github.com/source-academy/js-slang/pull/1361) to be merged and the js-slang version to be bumped to work as intended and so is marked as a draft until then.


### Type of change


### How to test
Checkout js-slang to https://github.com/source-academy/js-slang/pull/1361, and then [link local js-slang to local frontend](https://github.com/source-academy/js-slang#using-your-js-slang-in-your-local-source-academy), and start the frontend locally. The Source 4 Explicit Control variant should now be available as a selection. Switch to Source 4 Explicit Control and test by writing Source 4 code.
